ledger-tool/src/ledger_utils.rs

```diff
@@ -365,20 +365,36 @@ pub fn open_blockstore(
         },
     ) {
         Ok(blockstore) => blockstore,
-        Err(BlockstoreError::RocksDb(err))
-            if (err
+        Err(BlockstoreError::RocksDb(err)) => {
+            // Missing essential file, indicative of blockstore not existing
+            let missing_blockstore = err
                 .to_string()
-                // Missing column family
-                .starts_with("Invalid argument: Column family not found:")
-                || err
-                    .to_string()
-                    // Missing essential file, indicative of blockstore not existing
-                    .starts_with("IO error: No such file or directory:"))
-                && access_type == AccessType::Secondary =>
-        {
-            error!("Blockstore is incompatible with current software and requires updates");
+                .starts_with("IO error: No such file or directory:");
+            // Missing column in blockstore that is expected by software
+            let missing_column = err
+                .to_string()
+                .starts_with("Invalid argument: Column family not found:");
+            // The blockstore settings with Primary access can resolve the
+            // above issues automatically, so only emit the help messages
+            // if access type is Secondary
+            let is_secondary = access_type == AccessType::Secondary;
+
+            if missing_blockstore && is_secondary {
+                eprintln!(
+                    "Failed to open blockstore at {ledger_path:?}, it \
+                    is missing at least one critical file: {err:?}"
+                );
+            } else if missing_column && is_secondary {
+                eprintln!(
+                    "Failed to open blockstore at {ledger_path:?}, it \
+                    does not have all necessary columns: {err:?}"
+                );
+            } else {
+                eprintln!("Failed to open blockstore at {ledger_path:?}: {err:?}");
+                exit(1);
+            }
             if !force_update_to_open {
-                error!("Use --force-update-to-open to allow blockstore to update");
+                eprintln!("Use --force-update-to-open flag to attempt to update the blockstore");
```

Comment:
to me `error!()` is better, for I `grep ERROR` and that'd tell me what went wrong. `eprintln!()` assumes I know what I am looking for.

Reply:
Yeah, I'm inbetween on this.
- I personally like / am used to seeing `ERROR` ... the red text on terminal output and having `ERROR` to grep for are convenient
- I heard some feedback that the log statements look more "imposing" to a casual user whereas the regular `println!() / eprintln!()` are not.

But, I think that the feedback about logs is probably more pertinent to to users of `solana` CLI; `solana-ledger-tool` is primarily a tool used by people working on/with validator client, and as such, the logs are probably more familiar. Philosophically, I think `solana-ledger-tool` is more similar to `solana-validator` than it is to the CLI.

That being said, the rest of the `ledger-tool` currently uses `eprintln!()`, so the switch to `eprintln!()` makes these consistent. For the sake of this PR, I'd lean towards consistency, and then maybe in a subsequent PR, I can bump all the `eprintln!()` (or at least all the ones that make sense) to appropriate log (`error!()` / `warn!()` / etc). Thoughts ?

Reply:
consistency is always appreciated, plus 

>  it exits immediately after these lines 

makes good argument. 😄